apps/ejabberd/src/mod_amp.erl

```diff
@@ -45,21 +45,39 @@ stop(Host) ->
 
 %% Business API
 
--spec check_packet(exml:element(), amp_event()) -> exml:element() | drop.
+-spec check_packet(mongoose_acc:t() | exml:element(), amp_event()) ->
+    mongoose_acc:t() | exml:element() | drop.
 check_packet(Packet = #xmlel{attrs = Attrs}, Event) ->
+    % it is called this way only from ejabberd_c2s:send_and_maybe_buffer_stanza/3, line 1666
+    % maybe Paweł Chrząszcz knows why and can advise what to do about it
```

Comment:
@bartekgorny Could you contact @chrzaszcz directly? He may not get notifications so may not be aware of this question.